src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -1,9 +1,49 @@
 import argparse
+import os
 import re
+from pathlib import Path
 
 import pytest
 
-from diffpy.labpdfproc.tools import known_sources, set_wavelength
+from diffpy.labpdfproc.tools import known_sources, set_output_directory, set_wavelength
+
+params1 = [
+    ([None], ["."]),
+    (["."], ["."]),
+    (["new_dir"], ["new_dir"]),
+    (["existing_dir"], ["existing_dir"]),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params1)
+def test_set_output_directory(inputs, expected, tmp_path):
+    directory = Path(tmp_path)
+    os.chdir(directory)
+
+    existing_dir = Path().cwd().resolve() / "existing_dir"
+    existing_dir.mkdir(parents=True, exist_ok=True)
+
+    expected_output_directory = Path.cwd().resolve() / expected[0]
+    actual_args = argparse.Namespace(output_directory=inputs[0])
+    actual_args.output_directory = set_output_directory(actual_args)
+    assert actual_args.output_directory == expected_output_directory
+    assert Path(actual_args.output_directory).exists()
+    assert Path(actual_args.output_directory).is_dir()
+
+
+def test_set_output_directory_bad(tmp_path):
+    directory = Path(tmp_path)
+    os.chdir(directory)
+
+    existing_file = Path().cwd().resolve() / "existing_file.py"
```

Comment:
To make it less confusing let's just set this to `Path(tmp_path).resolve()` and not `Path().cwd().resolve()`